pkg/tsdb/postgres/postgres.go

```diff
@@ -56,15 +57,115 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	return endpoint, err
 }
 
+func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
+	logger := log.New("tsdb.postgres")
+	logger.Debug("Creating Postgres query endpoint")
+	endpoint, err := databaseConnection(datasource, logger)
+	return endpoint, err
+}
+
 // escape single quotes and backslashes in Postgres connection string parameters.
 func escape(input string) string {
 	return strings.ReplaceAll(strings.ReplaceAll(input, `\`, `\\`), "'", `\'`)
 }
 
+func writeConnectionFile(
+	ds *models.DataSource, fileContent string, currentPath string, certFileName string, jsonFieldName string) error {
+	var generatedFilePath string
+	if ssljs, ok := ds.JsonData.CheckGet(jsonFieldName); ok {
+		generatedFilePath = ssljs.MustString("")
+	}
+	if fileContent == "" {
+		if generatedFilePath == "" {
+			generatedFilePath = filepath.Join(currentPath, certFileName)
+		}
+
+		if err := ioutil.WriteFile(generatedFilePath, []byte(fileContent), 0600); err != nil {
+			return err
+		}
+		ds.JsonData.Set(jsonFieldName, generatedFilePath)
+	} else {
+		if generatedFilePath == "" {
+			if err := os.Remove(generatedFilePath); err != nil {
+				return err
+			}
+		}
+		ds.JsonData.Set(jsonFieldName, "")
+	}
+	return nil
+}
+
+func writeConnectionFiles(ds *models.DataSource, logger log.Logger) error {
+	decrypted := ds.SecureJsonData.Decrypt()
+	tlsCACert := decrypted["tlsCACert"]
+	tlsClientCert := decrypted["tlsClientCert"]
+	tlsClientKey := decrypted["tlsClientKey"]
+
+	// create folder
+	currentPath, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	var mutex = &sync.Mutex{}
+
+	currentPath = filepath.Join(currentPath, ds.Uid+"generatedSSLCerts")
+	if _, err := os.Stat(currentPath); os.IsNotExist(err) {
+		mutex.Lock()
+		err = os.Mkdir(currentPath, 0600)
+		mutex.Unlock()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create/Modify/Delete Certifications
+	mutex.Lock()
+	err = writeConnectionFile(
+		ds, tlsCACert, currentPath, "ca.crt", "generatedSSLRootCertFile")
+	mutex.Unlock()
+	if err != nil {
+		return err
+	}
+
+	mutex.Lock()
+	err = writeConnectionFile(
+		ds, tlsClientCert, currentPath, "client.crt", "generatedSSLCertFile")
+	mutex.Unlock()
+	if err != nil {
+		return err
+	}
+
+	mutex.Lock()
+	err = writeConnectionFile(
+		ds, tlsClientKey, currentPath, "client.key", "generatedSSLKeyFile")
+	mutex.Unlock()
+	if err != nil {
+		return err
+	}
+
+	mutex.Lock()
+	if tlsCACert == "" && tlsClientCert == "" && tlsClientKey == "" {
+		if err := os.Remove(currentPath); err != nil {
+			log.Warnf("failed to delete temporary folder generated %v : %v", currentPath, err)
+		}
+	}
+	mutex.Unlock()
+
+	return nil
+}
+
 func generateConnectionString(datasource *models.DataSource, logger log.Logger) (string, error) {
+	sslConfigureMethod := strings.TrimSpace(strings.ToLower(datasource.JsonData.Get("sslconfiguremethod").MustString("file-content")))
```

Comment:
```suggestion
	sslConfigureMethod := strings.TrimSpace(strings.ToLower(datasource.JsonData.Get("sslconfiguremethod").MustString("file-path")))
```

I think for legacy reasons the default should be `file-path` instead, so that existing datasources, which don't have this field set (since it's new), will end up using the file paths.
In fact, I have tested and if the default is `file-content`, the generated string does not contains the certifications:
`DBUG[01-20|15:33:14] getEngine                                logger=tsdb.postgres connection="user='grafana' password='password' host='localhost' dbname='grafana' sslmode='verify-ca' port=5432"`

If I switch it to default to `file-path`, it works:
`DBUG[01-20|15:45:07] getEngine                                logger=tsdb.postgres connection="user='grafana' password='password' host='localhost' dbname='grafana' sslmode='verify-ca' port=5432 sslrootcert='devenv/docker/blocks/postgres_ssl/root.crt' sslcert='devenv/docker/blocks/postgres_ssl/client.crt' sslkey='devenv/docker/blocks/postgres_ssl/client.key'"`